service/controller/v11/drainer_resource_set.go

```diff
@@ -26,7 +27,29 @@ func NewDrainerResourceSet(config DrainerResourceSetConfig) (*controller.Resourc
 	var err error
 
 	handlesFunc := func(obj interface{}) bool {
-		return true
+		p, err := key.ToPod(obj)
+		if err != nil {
+			return false
+		}
+		v, err := key.VersionBundleVersionFromPod(p)
+		// NOTE this is the hack we have to backport to ensure existing clusters
+		// work the way they do while paving the ground for v12 where we have to
+		// check more explictly against our desired version bundle version.
+		if v == "" || v == VersionBundle().Version {
+			return true
+		}
+
+		// TODO this error handling has to be enabled when going from v11 to v12.
+		//if err != nil {
+		//	return false
+		//}
+
+		// TODO this check has to be enabled when going from v11 to v12.
+		//if v == VersionBundle().Version {
+		//	return true
+		//}
+
+		return false
```

Comment:
All this here is not really nice but it works. In `v12` we have this fixed. Will introduce the next version in the coming PRs so we also have a WIP version bundle.